sql/core/benchmarks/StateStoreBasicOperationsBenchmark-results.txt

```diff
@@ -2,182 +2,109 @@
-RocksDB (trackTotalNumberOfRows: false)                                             10             10           0          1.0         979.8       0.7X
-
-OpenJDK 64-Bit Server VM 1.8.0_362-b09 on Linux 5.15.0-1035-azure
-Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
-evicting 7500 rows (maxTimestampToEvictInMillis: 7499) from 10000 rows:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
-------------------------------------------------------------------------------------------------------------------------------------------------------
-In-memory                                                                           6              7           0          1.6         610.0       1.0X
-RocksDB (trackTotalNumberOfRows: true)                                             34             35           0          0.3        3402.0       0.2X
-RocksDB (trackTotalNumberOfRows: false)                                             8              9           0          1.2         825.0       0.7X
+In-memory                                                                            6              7           0          1.6         625.7       1.0X
+RocksDB (trackTotalNumberOfRows: true)                                              60             61           2          0.2        5950.1       0.1X
+RocksDB (trackTotalNumberOfRows: false)                                             22             23           1          0.5        2194.1       0.3X
 
-OpenJDK 64-Bit Server VM 1.8.0_362-b09 on Linux 5.15.0-1035-azure
-Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
+OpenJDK 64-Bit Server VM 1.8.0_362-b09 on Linux 5.15.0-1037-azure
+Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
 evicting 5000 rows (maxTimestampToEvictInMillis: 4999) from 10000 rows:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------------------------------------
-In-memory                                                                           6              6           0          1.8         565.1       1.0X
-RocksDB (trackTotalNumberOfRows: true)                                             24             25           0          0.4        2383.2       0.2X
-RocksDB (trackTotalNumberOfRows: false)                                             7              7           0          1.5         665.8       0.8X
+In-memory                                                                           6              7           1          1.7         586.7       1.0X
+RocksDB (trackTotalNumberOfRows: true)                                             31             33           1          0.3        3146.3       0.2X
+RocksDB (trackTotalNumberOfRows: false)                                            13             13           0          0.8        1260.5       0.5X
```

Comment:
The change is ok to me, but please confirm again: The `Per Row(ns)` of  `RocksDB (trackTotalNumberOfRows: false)` all seem to be twice the original, is this indeed expected? @HeartSaVioR @anishshri-db

Reply:
Yes that is expected since the representation for writeBatch and memtables is not exactly the same. As mentioned earlier, overall there is less impact since we can avoid the write in the commit phase which is not captured here. Together with the memory changes we already have and the changelog checkpointing change (https://github.com/apache/spark/pull/41099) in review, we can achieve bounded memory usage as well as better overall performance.